core/domain/stats_services.py

```diff
@@ -136,25 +136,24 @@ def update_stats(exp_id, exp_version, aggregated_stats):
     exploration_stats.num_actual_starts_v2 += aggregated_stats[
         'num_actual_starts']
 
-    for state_name in aggregated_stats['state_stats_mapping']:
-        exploration_stats.state_stats_mapping[
-            state_name].total_answers_count_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['total_answers_count']
-        exploration_stats.state_stats_mapping[
-            state_name].useful_feedback_count_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['useful_feedback_count']
-        exploration_stats.state_stats_mapping[
-            state_name].total_hit_count_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['total_hit_count']
-        exploration_stats.state_stats_mapping[
-            state_name].first_hit_count_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['first_hit_count']
-        exploration_stats.state_stats_mapping[
-            state_name].num_times_solution_viewed_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['num_times_solution_viewed']
-        exploration_stats.state_stats_mapping[
-            state_name].num_completions_v2 += aggregated_stats[
-                'state_stats_mapping'][state_name]['num_completions']
+    for state_name, aggregated_state_stats in (
+            aggregated_stats['state_stats_mapping'].items()):
+        if state_name not in exploration_stats.state_stats_mapping:
+            exploration_stats.state_stats_mapping[state_name] = (
+                stats_domain.StateStats.create_default())
```

Comment:
Rather than create default, will it be possible to generate the correct stats and commit that to the datastore. I feel that things like the views, completions, etc for that state is lost right now and I feel at least by commiting a model to the datastore, we can count upcoming events correctly... Did I not understand this correctly?

Reply:
Hmm, I think it might be possible if we search "backwards" for the same state in storage. However, if it still doesn't exist, then that means we'd need to create a zero-state anyway. WDYT?

/cc @seanlip @kevintab95

Reply:
I think that sounds reasonable. As long as we are recovering as much data as possible, I think it is fine. And where we have no data, of course, we need to restart.

Reply:
Though, it might also help to investigate why these stats models are missing in the first place....